### PR TITLE
fix: catch && handle plugin `ValidationErrors` (`schema-utils`)

### PR DESCRIPTION
**Pull request by michael-ciniawsky**
**What kind of change does this PR introduce?**

- [x] Bugfix 

**Did you add tests for your changes?**

- [ ] TODO (if backporting is approved => requires an update of one `devDependencies` plugin e.g ETWP) 

**If relevant, link to documentation update:**

- N/A

**Summary**

This is needed for contrib being able to use the lastest version of `schema-utils` for plugins without relying on the ugly `process.exit(1)` hack in `schema-utils =< v0.4.1` 

The changes are a backport of #5892  for `webpack >= v3.0.0` (`master`), so if this commit [`6d7df9d`](https://github.com/webpack/webpack/pull/5892/commits/6d7df9daaddac8976539884294b871e8e83f8bde) can be cherry-picked from the `next` branch instead this PR can be closed :)

**Does this PR introduce a breaking change?**

I hope not 🙃

### CODE REVIEW

**Comment by sokra** [comment]
``` text
/home/travis/build/webpack/webpack/lib/ErrorHelpers.js
  50:2  error  Missing semicolon  semi
```

**Comment by sokra** [comment]
The CI got pretty unstable in the last days...

**Comment by sokra** [comment]
Could be related to this:

Travis:
Nov 28, 2017
Container-based Linux reduced capacity

**Comment by michael-ciniawsky** [comment]
Yep, when `schema-utils` is fixed I update EWTP and add a test

**Comment by sokra** [comment]
@ev1stensberg

**Comment by michael-ciniawsky** [comment]
@sokra Will the CLI be backported to `webpack v3.0.0` aswell, since this PR is a backport of https://github.com/webpack/webpack/pull/5892/commits/6d7df9daaddac8976539884294b871e8e83f8bde (`webpack v4.0.0`)? But in any case this should be upstreamed to `webpack-cli` aswell, I will try but hopefully the logic didn't change too much there... 😅 

Still waiting for a `schema-utils` release to add a test for this 😞...

File: lib/ErrorHelpers.js [hunk 1]
```diff
@@ -6,14 +6,27 @@
...
 
+exports.cutOffLoaderExecution = (stack) => exports.cutOffByFlag(stack, loaderFlag);
+
+exports.cutOffWebpackOptinos = (stack) => exports.cutOffByFlag(stack, webpackOptionsFlag);
+
+exports.cutOffMultilineMessage = (stack, message) => {
+	stack = stack.split("\n");
+	message = message.split("\n");
+
+	return stack.reduce((acc, line, idx) => line.indexOf(message[idx]) < 0 ? acc.concat(line) : acc, []).join("\n");
```
[the remark this answers is not in the document]
Comment (michael-ciniawsky) [reply lib/ErrorHelpers.js]: done :)

**Comment by ev1stensberg** [comment]
@michael-ciniawsky happy to help you porting it to the CLI, sorry for late reply.

**Comment by michael-ciniawsky** [comment]
@ev1stensberg You can either pick the changes here or I will try to make a PR as soon as possible this still is on hold waiting for a `schema-utils` release, so I can add a test :)

**Comment by ev1stensberg** [comment]
@michael-ciniawsky Could you do a PR? I'm bad at picking.. 😢

File: lib/ErrorHelpers.js [hunk 2]
```diff
@@ -29,3 +44,9 @@ exports.cleanUp = (stack, message) => {
 	stack = exports.cutOffMessage(stack, message);
 	return stack;
 };
+
+exports.cleanUpWebpackOptions = (stack, message) => {
+	stack = exports.cutOffWebpackOptinos(stack);
```
Comment (ooflorent) [inline lib/ErrorHelpers.js]: There is a typo here.

**Comment by TheLarkInn** [comment]
Is this still targeting our v4 release timeline? If not I'm happy to move this to v4.x (or someone can go ahead and do so).

**Comment by ev1stensberg** [comment]
I think maybe this got fixed when we added the CLI as an encapsulated function

**Comment by michael-ciniawsky** [comment]
This was intended to be an explicit backport for `webpack =< 4.0.0` not sure if still relevant (I lost track here tbh :)) for `schema-utils` (Options Validation) I think

**Comment by sokra** [comment]
this was merged into webpack-cli

**Pull request closed without merging**